hal/common/retarget.cpp

```diff
@@ -713,6 +713,37 @@ extern "C" void __env_unlock( struct _reent *_r )
 {
     __rtos_env_unlock(_r);
 }
+
+extern "C" int __cxa_guard_acquire (uint64_t *guard_object)
+{
```

Comment:
nit: Rather than casting every access of the guard_object, why not cast once per function?
``` cpp
extern "C" int __cxa_guard_acquire (uint64_t *guard_object_p)
{
    uint8_t *gaurd_object =(uint8_t *)p;
    if (*gaurd_object) {
        ...
```